lib/std/json/static.zig

```diff
@@ -42,6 +42,12 @@ pub const ParseOptions = struct {
     /// The default with a `*std.json.Reader` input is `.alloc_always`.
     /// Ignored for `parseFromValue` and `parseFromValueLeaky`.
     allocate: ?AllocWhen = null,
+
+    /// Determines whether or not `std.json.Token.number` and `std.json.Token.allocated_number`, values are parsed.
+    /// When parsing to a `std.json.Value`; if this boolean is false,
+    /// then numbers that would otherwise be parsed into either a `std.json.Value.integer`
+    /// or `std.json.Value.float` will remain unparsed as a `std.json.Value.number_string`.
```

Comment:
`.number_string` is actually used sometimes when this option is true. Here's a suggestion for how to word this new option:
```suggestion
    /// When parsing to a `std.json.Value`, set this option to false to always emit
    /// JSON numbers as unparsed `std.json.Value.number_string`.
    /// Otherwise, JSON numbers are parsed as either `std.json.Value.integer`
    /// or `std.json.Value.float` or left as unparsed `std.json.Value.number_string`
    /// depending on the format and value of the JSON number.
    /// When this option is true, JSON numbers encoded as floats (see `std.json.isNumberFormattedLikeAnInteger`)
    /// may lose precision when being parsed into `std.json.Value.float`.
```